lightning-persister/src/lib.rs

```diff
@@ -32,6 +32,22 @@ use std::io::{Cursor, Error};
 use std::ops::Deref;
 use std::path::{Path, PathBuf};
 
+impl DiskWriteable for Vec<u8> {
```

Comment:
Would strongly prefer to not encode into a vec and then dump, instead we can just define `persist_bytes` as generic over a `DiskWriteable`.

Reply:
Hm, but then it's only intended to be used by a FilesystemPersister. Part of the motivation for this is to make it easier to implement a generic persistence layer.

Reply:
I mean we can also make the `DiskWriteable` trait public and define it in terms of a `Writer`?

Reply:
Or, really, we could drop `DiskWriteable` entirely and use the `lightning` crate's existing `Writeable` trait.

Reply:
nice. removed DiskWriteable and was able to use Writeable everywhere instead and avoid the extra encode()